nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -410,3 +412,27 @@ def test__trim_maps():
     np.testing.assert_equal(mask_data, maps_i_mask)
     mask_data[1, 1, 1] = 1  # reset, just in case.
     np.testing.assert_equal(np.asarray(list(range(4))), maps_i_indices)
+
+
+@pytest.mark.parametrize('target_dtype',
+                         (np.float, np.float32, np.float64, np.int, np.uint),
+                         )
+def test_img_to_signals_labels_int_type(target_dtype):
+    fake_fmri_data = np.random.RandomState(0).rand(10, 10, 10, 10) > 0.5
+    fake_affine = np.random.rand(4, 4).astype(np.float64)
```

Comment:
why a random affine? use `np.eye`. also, always use a fixed seed in tests, not `np.random.rand`

Reply:
Isn't that what `RandomState(0)` is doing?

FMI, what's better about np.eye instead of a random affine?